### ISSUE TITLE
Redirect the browser window instead of using `open`

### ISSUE THREAD

**Issue by aaronpk** [bitfocus/companion-module-youtube-live#67]
Can this start the Google OAuth flow by redirecting the actual Companion browser accessing the web interface instead of trying to launch a new browser?

I and many others run Companion on a separate server like a Raspberry Pi, which has no GUI on it, so the node `open` command doesn't do anything.

I access the Companion interface over the network, so this should start the OAuth flow in that same browser. All it would need to do is set the `window.location` to the `consentScreenUrl` instead of use the `opn` function.

In the mean time, I was able to work around this by grabbing the `consentScreenUrl` from the Companion Log, visiting that in my browser, then taking the redirect URL from Google and running `curl http://localhost:3000/?code=xxxx` from the command line on the Pi.

**Comment by TerrrorByte** [comment]
@aaronpk How were you able to get the `consentScreenUrl` from the log? When I open my log I do not see that, and on my iMac it's not opening a new browser or tab, so I'm not able to authenticate at all :(

**Comment by Julusian** [comment]
I agree that using `opn` is not a great idea, as many users are using companionpi or configuring companion over the network.  
But modules are not able to do a redirect in the browser.
Some other modules doing oauth handle this by simply outputting the url as a config field, requiring the person configuring to copy and go that url.
This is probably something companion core should think about supporting better for modules

**Comment by lfrisk** [comment]
I too ran into this authentication roadblock. I was running CompanionPi 3.1.2, and saw that 3.2.1 was available, hoping that this was fixed by that release...which it has not been.
Is there a workaround to get this to authenticate on CompanionPi?

**Comment by lfrisk** [comment]
I tried following the workaround that the original poster mentioned about copying the consent url...can you please break this process down a bit?  I was able to grab a url from the log...I used the command line of an ssh session to run this url...and I did not get any usable results...maybe I don't know what I am looking for?
